### ISSUE TITLE
Flow: Specify start and end time for pathline injection

### ISSUE THREAD

**Issue by sgpearse** [NCAR/VAPOR#3327]
The Flow/pathline renderer currently allows for an Injection Interval, but the user cannot specify the timespan on which the seeds are applied.  If a user has a dataset with many time steps, but is only interested in timesteps later on in the simulation, they are forced to integrate flow for all timesteps until they get to the timespan of interest.

It would be really nice to have parameters for "Pathline Injection Start" and "Pathline Injection End" like what was available in Vapor 2.

**Comment by shaomeng** [comment]
> If a user has a dataset with many time steps, but is only interested in timesteps later on in the simulation, they are forced to integrate flow for all timesteps until they get to the timespan of interest.
> 
Is it also an option to load only the timespan of interest, but not the whole thing?

**Comment by sgpearse** [comment]
That's a workaround, but it adds burden to the user. IMO, it isn't friendly to make users render three different sessions to 1) render before flow, 2) render during flow, and 3) render after flow.  Vapor 2 had these parameters for this reason.

**Comment by shaomeng** [comment]
> That's a workaround, but it adds burden to the user. IMO, it isn't friendly to make users render three different sessions to 1) render before flow, 2) render during flow, and 3) render after flow. Vapor 2 had these parameters for this reason.

I hear you, it's gonna be an useful addition. From a cost-benefit perspective, the added value is quite minimal, while the cost is non-trivial. After all, I consider it a *very* reasonable request for users, (or maybe even an encouraged way of interacting with large data sets,) to load only data that they're interested in.

**Comment by sgpearse** [comment]
Ok, let's chat about this in our meeting or at some other point.  But here are my thoughts.

> From a cost-benefit perspective, the added value is quite minimal

I have to disagree.  This feature was in Vapor 2 for a good reason.  In my use case, I have six renderers in my session.  Without being able to specify when the flow renderer advects, I need to create two new sessions (one before pathlines and one after) that contain six new renderers that are identical to those in original session.  

This is tedious and prone to error.  If I get a parameter wrong in one of my 18 renderers I will have an inconsistency in my visualization.  This problem gets worse if I have more than six renderers, or multiple time spans that I want to integrate.

> while the cost is non-trivial

Can't this be achieved with a simple condition in _paintGL()?  IE

```
int FlowRenderer::_paintGL() {
    auto rp = GetActiveParams();
    if (!rp->GetIsSteady()) {
        auto currentTimestep = rp->GetCurrentTimestep();
        auto startTime = rp->GetStartTime();
        auto endTime = rp->GetEndTime();
        if ( currentTimestep < startTime || currentTimestep > endTime ) return 0;
    }
...
```

> After all, I consider it a very reasonable request for users, (or maybe even an encouraged way of interacting with large data sets,) to load only data that they're interested in.

In most cases, the data of interest is not known a priori.  This is Vapor's primary use case, which is to explore the data.

**Comment by shaomeng** [comment]
> > After all, I consider it a very reasonable request for users, (or maybe even an encouraged way of interacting with large data sets,) to load only data that they're interested in.
> 
> In most cases, the data of interest is not known a priori. This is Vapor's primary use case, which is to explore the data.

@sgpearse  I think you're confusing different use cases. If the data of interest is not known a priori, the feature you requested cannot help either. If the interesting region is known a priori, either the feature you requested or selectively loading a subset of data can help. Make sense?

**Comment by sgpearse** [comment]
> If the data of interest is not known a priori, the feature you requested cannot help either.

No, that's incorrect.  This feature would help tremendously for my use case as it has in the past.  The use case is as follows:

1) Load the entirety of a simulation's output
2) Explore the entire time series and identify the time steps with the largest TKE values in a specific location
3) Enable pathline renderings only during these select times

If this isn't a useful feature it would not have been implemented in Vapor 2.

![Screen Shot 2023-02-07 at 10 55 07 AM](https://user-images.githubusercontent.com/9522770/217326963-2c2011bc-83e6-4e1c-9a6c-19d3ab06031d.png)
